db/schema.rb

```diff
@@ -182,6 +180,16 @@
     t.index ["resource_id"], name: "index_pages_on_resource_id"
   end
 
+  create_table "resource_tool_groups", force: :cascade do |t|
+    t.bigint "resource_id", null: false
```

Comment:
The docs say,

> The bigint type is designed to be used when the range of the integer type is insufficient

and

> The type integer is the common choice, as it offers the best balance between range, storage size, and performance

So, if you don't need to store numbers > 2B, stick with `t.integer`.

On the other hand, this merely follows suit from the existing `resource_id` in `language_attributes` - but I question the need for that bigint as well. These are all foreign keys to IDs that are only int4 - plain integer.

Reply:
@andrewroth @knutsenm 

I think that this is a default behaviour of RoR migrations, although ofcourse we can convertet them into Integer.

If we don't need this bigint, should we consider to convert to this to `Integer` type in both tables `resource_tool_groups` and `language_attributes` column `resource_id`?

Reply:
That sounds good to me.

Reply:
@andrewroth  
Good, then I will convert column type for both tables

Reply:
> I think that this is a default behaviour of RoR migrations

I doubt it, but I'm not positive about that. In production, resources.id is an `int4` - i.e. a plain `integer`, so the `t.references :resource` should have generated an `integer` here.

Perhaps your local database has a bigint for `resources.id` ? Which version of PostgreSQL are you running? We're on v11 (which surprises me - we've moved to v14 most everywhere)

Reply:
Done

Reply:
You were right: Rails defaults to bigint for new primary keys since 5.1: https://github.com/rails/rails/pull/26266

That doesn't explain, however, why this reference to an existing integer became a bigint.

Reply:
> You were right: Rails defaults to bigint for new primary keys since 5.1: [rails/rails#26266](https://github.com/rails/rails/pull/26266)
> 
> That doesn't explain, however, why this reference to an existing integer became a bigint.
@knutsenm 
I think this is because when core team though about this change, it was with the intention to implement from there to the future the bigint for a reference no matter if this was represented as integer cuz they think that is better to cover this possibility to migrate later just the reference origin type and not the entire relationship, I mean, just adjust the origin id in table and not all the things.. origin and reference value.

I think the main idea is to prevent future big work in migrations, against a low cost of performance now cuz the databases are addapting each time better to work with big numbers without a big impact in performance